language_detector/main.py

```diff
@@ -1,9 +1,21 @@
+# -​*- coding: utf-8 -*​-
+
 """This is the entry point of the program."""
 
 from .languages import LANGUAGES
 
-
 def detect_language(text, languages=LANGUAGES):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+
+    input_text = text.split() 
+
+    counter_dict = {'spanish': 0, 'german': 0, 'english': 0}
+
+    for language in languages:
```

Comment:
Currently, the for loop goes through each "common word" from the imported list once a language is selected. Each word is iterating through each word of the given text. This means iterations this program will go through will be (# of languages) * (# of common words) * (# of words in the given text). I think this iteration number can go up pretty fast if each parameter grows. I think the use of Set which is efficient at finding a unique element would be helpful.